evm_loader/program/src/types/transaction.rs

```diff
@@ -117,6 +135,299 @@ impl rlp::Decodable for Transaction {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct AccessListTx {
+    nonce: u64,
+    gas_price: U256,
+    gas_limit: U256,
+    target: Option<Address>,
+    value: U256,
+    call_data: crate::evm::Buffer,
+    r: U256,
+    s: U256,
+    chain_id: U256,
+    recovery_id: u8,
+    access_list: Vec<(Address, Vec<StorageKey>)>,
+    rlp_len: usize,
+    hash: [u8; 32],
+    signed_hash: [u8; 32],
```

Comment:
`rlp_len, hash and signed_hash` technically are not part of the transaction, we need to calculate them for all transaction types and they could be moved outside.

Like:
```
enum TransactionType {
    Legacy(LegacyTx),
    AccessList(AccessListTx),
}

pub struct Transaction {
   tx: TransactionType,
   rlp_len: usize,
   hash: [u8; 32],
   signed_hash: [u8; 32],
}
```

Reply:
extracted to a separate struct